libraries/core_libs/consensus/include/final_chain/final_chain.hpp

```diff
@@ -131,7 +131,11 @@ class FinalChain {
   std::optional<TransactionReceipt> transactionReceipt(EthBlockNumber blk_n, uint64_t position,
                                                        std::optional<trx_hash_t> trx_hash = {}) const;
 
-  /**
+  std::shared_ptr<Transaction> transaction(EthBlockNumber blk_n, uint32_t position) const {
+    return db_->getTransaction(blk_n, position);
```

Comment:
I think it worth to checkif period is in trasnactions_cache_

Reply:
Done